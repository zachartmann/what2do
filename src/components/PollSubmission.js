import React, { useState } from "react";
import IncludeName from "../components/IncludeName";
import Info from "../components/Info";
import { postPoll } from "../common/requests/Poll";

const PollSubmission = (props) => {
  const [question, setQuestion] = useState("");
  const [selectedTemplate, setSelectedTemplate] = useState("what2do");
  const [selectedDuration, setSelectedDuration] = useState("60"); // in minutes
  const templates = ["what2do", "what2play", "where2go"];
  const times = [
    "10 min",
    "30 min",
    "1 hour",
    "2 hours",
    "4 hours",
    "6 hours",
    "12 hours",
    "1 day",
    "2 days",
    "3 days",
    "5 days",
    "10 days",
    "1 week",
    "2 weeks",
  ];
  const timesTemp = ["10", "30", "60"];
  const placeholders = {
    what2do: "What should we do?",
    what2play: "What should we play?",
    where2go: "Where should we go?",
  };

  const changeTemplate = (event) => {
    setSelectedTemplate(event.target.value);
  };

  function generatePollId() {
    let pollId = "";
    for (let i = 0; i < 4; i++) {
      pollId += String.fromCharCode(65 + Math.floor(Math.random() * 26));
    }
    // TODO: check if poll ID already exists
    return pollId;
  }

  const handleSubmit = (evt) => {
    evt.preventDefault();
    const pollId = generatePollId();
    alert(
      `Submitting ${pollId}\nTemplate: ${selectedTemplate}\nQuestion: ${question}\nDuration: ${selectedDuration}\nEnd: ${new Date(
        new Date().getTime() + selectedDuration * 60000
      )}`
    );
    postPoll(
      pollId,
      question,
      new Date().getTime() + selectedDuration * 60000,
      selectedDuration,
      "ideaId" /*TODO: initial ideas from template*/
    );
  };

  return (
    <div className="content">
      <div className="content-container">
        <div className="content-container">
          <h3>What do you want to ask?</h3>
        </div>
        <form onSubmit={handleSubmit}>
          <div className="content-container">
            <select
              name="template"
              id="template"
              value={selectedTemplate}
              onChange={changeTemplate}
            >
              {templates.map((template) => {
                return (
                  <option key={template} value={template}>
                    {template}
                  </option>
                );
              })}
            </select>
          </div>
          <div className="content-container">
            <p>Question</p>
          </div>
          <div className="content-container">
            <input
              placeholder={placeholders[selectedTemplate]}
              onChange={(e) => setQuestion(e.target.value)}
            />
          </div>
          <div className="content-container">
            <select
              name="template"
              id="template"
              value={selectedDuration}
              onChange={(e) => setSelectedDuration(e.target.value)}
            >
              {timesTemp.map((time) => {
                return (
                  <option key={time} value={time}>
                    {time}
                  </option>
                );
              })}
            </select>
          </div>
          <div className="content-container flex-container">
            <div className="flex-component flex-70">
              <IncludeName />
              <Info text="Give us your details." />
            </div>
            <div className="flex-component flex-30 flex-end">
              <button type="submit">Send</button>
            </div>
          </div>
        </form>
      </div>
    </div>
  );
};

export default PollSubmission;
