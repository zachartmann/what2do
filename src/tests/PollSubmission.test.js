import { shallow } from "enzyme";
import PollSubmission from "../components/PollSubmission";

describe("PollSubmission component", () => {
  it("should render correctly with given idea", () => {
    const component = shallow(<PollSubmission />);

    expect(component).toMatchSnapshot();
  });
});
