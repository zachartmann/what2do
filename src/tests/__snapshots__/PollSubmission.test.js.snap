// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PollSubmission component should render correctly with given idea 1`] = `
<div
  className="content"
>
  <div
    className="content-container"
  >
    <div
      className="content-container"
    >
      <h3>
        What do you want to ask?
      </h3>
    </div>
    <div
      className="content-container"
    >
      <p>
        Template
      </p>
    </div>
    <form
      onSubmit={[Function]}
    >
      <div
        className="content-container"
      >
        <select
          id="template"
          name="template"
          onChange={[Function]}
          value="None"
        >
          <option
            key="None"
            value="None"
          >
            None
          </option>
        </select>
      </div>
      <div
        className="content-container"
      >
        <p>
          Question
        </p>
      </div>
      <div
        className="content-container"
      >
        <input
          onChange={[Function]}
          placeholder="Enter a question!"
        />
      </div>
      <div
        className="content-container"
      >
        <select
          id="template"
          name="template"
          onChange={[Function]}
          value="60"
        >
          <option
            key="10"
            value="10"
          >
            10
          </option>
          <option
            key="30"
            value="30"
          >
            30
          </option>
          <option
            key="60"
            value="60"
          >
            60
          </option>
        </select>
      </div>
      <div
        className="content-container flex-container"
      >
        <div
          className="flex-component flex-70"
        >
          <IncludeName />
          <Info
            text="Give us your details."
          />
        </div>
        <div
          className="flex-component flex-30 flex-end"
        >
          <button
            type="submit"
          >
            Send
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
